### DISCUSSION TITLE
failure to disable SID

### DISCUSSION THREAD

**Discussion by wtl06639** [Security-Onion-Solutions/securityonion#3282] [Unsupported Versions]
I have looked over a few of the discussions and following the SO 2.3 documentation but it does not seem to disable the SID for me.

I have place the following in the global.sls file on the manager.  and  I am currently testing with just one heavy node. I can run salt <manager>_<role> state.apply idstools and it runs successful since. Sensor isnt running idstools so i cant run for that. But it i check and the sensor is still generating alerts for the SIDS. 

idstools:
  sids:
    enabled:
    disabled:
      - 38747
      - 38748
      - 38749
      - 38750
    modify:

**Comment by Grey0wl** [comment]
I am working on a similar issue disabling rules globally doesn't seem to work. I have a distributed environment as well and my manager node is a manager/search node.  I also tried removing the disable rules from the global.sls file and tried to disable for a specific node only by adding the disabled filter to the /pillar/minion/$Minion.sls in the below format with no luck, have you tried applying the rule directly to a minion? I am curious if it works for you. 

idstools:
  config:
    ruleset: 'Emerging Threats'
    oinkcode: ''
    urls:
  sids:
    enabled:
    disabled:
    - 2027766
    modify:

**Comment by TOoSmOotH** [comment]
Please try and use so-rule. It was added in 2.3.30 for disabling rules. The only place you change rules is the manager as its where the rule management takes place. 

So to disable the sid 2027766 I would do the following:

`sudo so-rule disabled add 2027766 
`

For good measure follow it up with:

`sudo so-rule-update
`

Apply it to all sensors if distrib:

`sudo salt "*_sen*" state.apply suricata
`

**Reply by wtl06639** [reply]
I updated and was able to run that command and it looks like it took the command but i am still see the alerts i disabled still being generated. I check the all.rules files and they are not # out. not sure if thats where it will set the disable or not

**Reply by TOoSmOotH** [reply]
I just confirmed you have to run so-rule-update for it to take. 

I see it commented out when I grep now

**Reply by wtl06639** [reply]
Nope. I had run that already. Still not showing commented out. unless im doing it wrong:

[PC user]# so-rule disabled add 38747
38747 already exists in disabled
[PC user]# so-rule-update
2021-03-04 21:55:47,498 - <INFO> - Loading ./rulecat.conf.
2021-03-04 21:55:47,500 - <INFO> - Forcing Suricata version to 6.0.
2021-03-04 21:55:47,500 - <INFO> - Last download less than 15 minutes ago. Not downloading https://www.snort.org/rules/snortrules-snapshot-2983.tar.gz?oinkcode=85c5e131655f2d8bb190d0a37edf526395d0815e.
2021-03-04 21:55:51,463 - <INFO> - Ignoring file rules/deleted.rules
2021-03-04 21:55:51,468 - <INFO> - Loading local file /opt/so/rules/nids/local.rules
2021-03-04 21:55:51,468 - <INFO> - Loading local file /opt/so/rules/nids/sorules/extraction.rules
2021-03-04 21:55:51,468 - <INFO> - Loading local file /opt/so/rules/nids/sorules/filters.rules
2021-03-04 21:55:54,283 - <INFO> - Loaded 44752 rules.
2021-03-04 21:55:54,552 - <INFO> - Disabled 0 rules.
2021-03-04 21:55:54,552 - <INFO> - Enabled 0 rules.
2021-03-04 21:55:54,552 - <INFO> - Modified 0 rules.
2021-03-04 21:55:54,552 - <INFO> - Dropped 0 rules.
2021-03-04 21:55:54,667 - <INFO> - Enabled 0 rules for flowbit dependencies.
2021-03-04 21:55:58,163 - <INFO> - Writing rules to /opt/so/rules/nids/all.rules: total: 44752; enabled: 12620; added: 0; removed 0; modified: 0
2021-03-04 21:55:58,554 - <INFO> - No changes detected, will not reload rules or run post-hooks.
[PC user]# more /opt/so/rules/nids/all.rules | grep 38747
alert tcp $HOME_NET any -> $EXTERNAL_NET any (msg:"MALWARE-CNC FF-RAT outbound connection attempt"; sid:38747; gid:3; rev:1; classtype:trojan-activity; metadata: engine shared, soid 3|38747, policy balanced-ips drop, policy security-ips drop, impact_flag red;)
[PC user]#

**Comment by joeb1kenobe** [comment]
Could it be because this is a stub rule for a precompiled talos detection.
This has a GID of 3 instead of 1.

Joe

On Thu, Mar 4, 2021 at 4:58 PM wtl06639 <notifications@github.com> wrote:

> Nope. I had run that already. Still not showing commented out. unless im
> doing it wrong:
>
> [PC user]# so-rule disabled add 38747
> 38747 already exists in disabled
> [PC user]# so-rule-update
> 2021-03-04 21:55:47,498 - - Loading ./rulecat.conf.
> 2021-03-04 21:55:47,500 - - Forcing Suricata version to 6.0.
> 2021-03-04 21:55:47,500 - - Last download less than 15 minutes ago. Not
> downloading
> https://www.snort.org/rules/snortrules-snapshot-2983.tar.gz?oinkcode=85c5e131655f2d8bb190d0a37edf526395d0815e
> .
> 2021-03-04 21:55:51,463 - - Ignoring file rules/deleted.rules
> 2021-03-04 21:55:51,468 - - Loading local file
> /opt/so/rules/nids/local.rules
> 2021-03-04 21:55:51,468 - - Loading local file
> /opt/so/rules/nids/sorules/extraction.rules
> 2021-03-04 21:55:51,468 - - Loading local file
> /opt/so/rules/nids/sorules/filters.rules
> 2021-03-04 21:55:54,283 - - Loaded 44752 rules.
> 2021-03-04 21:55:54,552 - - Disabled 0 rules.
> 2021-03-04 21:55:54,552 - - Enabled 0 rules.
> 2021-03-04 21:55:54,552 - - Modified 0 rules.
> 2021-03-04 21:55:54,552 - - Dropped 0 rules.
> 2021-03-04 21:55:54,667 - - Enabled 0 rules for flowbit dependencies.
> 2021-03-04 21:55:58,163 - - Writing rules to /opt/so/rules/nids/all.rules:
> total: 44752; enabled: 12620; added: 0; removed 0; modified: 0
> 2021-03-04 21:55:58,554 - - No changes detected, will not reload rules or
> run post-hooks.
> [PC user]# more /opt/so/rules/nids/all.rules | grep 38747
> alert tcp $HOME_NET any -> $EXTERNAL_NET any (msg:"MALWARE-CNC FF-RAT
> outbound connection attempt"; sid:38747; gid:3; rev:1;
> classtype:trojan-activity; metadata: engine shared, soid 3|38747, policy
> balanced-ips drop, policy security-ips drop, impact_flag red;)
> [PC user]#
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Security-Onion-Solutions/securityonion/discussions/3282#discussioncomment-431014>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAD6W6RICWYIUJREZFFSLP3TB766TANCNFSM4YTM6TAA>
> .
>
-- 
Joe Brown

**Reply by wtl06639** [reply]
Is there anyway to disable it if it is a stub rule?
